torch/csrc/jit/codegen/cuda/arith.cpp

```diff
@@ -88,24 +131,29 @@ TensorView* newOutputTV(const std::vector<Val*>& vals, DataType dtype) {
         out_domain.size());
     for (const auto i : c10::irange(dom.size())) {
       if (dom[i]->isBroadcast()) {
+        if (dom[i]->hasExpandedExtent()) {
+          expanded_extent_vals[i] =
+              promoteSize(expanded_extent_vals[i], dom[i]->expandedExtent());
+        }
         continue;
       }
-      if (extent_vals[i] == nullptr) {
-        extent_vals[i] = dom[i]->extent();
-        iter_types[i] = dom[i]->getIterType();
-      }
+      extent_vals[i] = promoteSize(extent_vals[i], dom[i]->extent());
+      iter_types[i] = dom[i]->getIterType();
```

Comment:
I left a note, but we can't do this now because of gather. It seems to me gather can be promoted to `IterType::Serial` pretty aggressively, but one thing we should do, is if people gather with a window size of 1, shouldn't we just make that an explicit broadcast? Is this what `convertToBroadcast` was doing? If so, that should just be done in gather definition.